### PR TITLE
Cav 321 automate what the passcode is for the acceptance tests

### PR DESCRIPTION
**Pull request by brdgeth**
Added a get otp endpoint that is called by the acceptance tests repo only to enable automation of the happy path for verification.
Get otp gets the expected passcode for a given phone number.
The expected passcode is put in a mongodb cache during the verify call.

### CODE REVIEW

File: app/uk/gov/hmrc/cipphonenumberstubs/dao/OtpCacheRepository.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import uk.gov.hmrc.mongo.cache.{CacheIdType, MongoCacheRepository}
+
+import javax.inject.Inject
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.FiniteDuration
+
+class OtpCacheRepository @Inject()(mongoComponent: MongoComponent,
+                                   configuration: Configuration,
+                                   timestampSupport: TimestampSupport)(implicit ec: ExecutionContext)
+  extends MongoCacheRepository(
```
Comment (mar36tin) [inline app/uk/gov/hmrc/cipphonenumberstubs/dao/OtpCacheRepository.scala:30]: We should be using the **[DefaultPlayMongoRepositorySupport](https://github.com/hmrc/hmrc-mongo/tree/main/hmrc-mongo-test-play-28)** as this allows us to setup and tear down after tests. That way we don't have a Mongocache that stays on even when tests are not running

**Review by mar36tin — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/cipphonenumberstubs/dao/OtpCacheRepository.scala (hunk 1 above)
Comment (brdgeth) [reply app/uk/gov/hmrc/cipphonenumberstubs/dao/OtpCacheRepository.scala:30]: ok i will make another ticket

Comment (brdgeth) [reply app/uk/gov/hmrc/cipphonenumberstubs/dao/OtpCacheRepository.scala:30]: please see https://jira.tools.tax.service.gov.uk/browse/CAV-379
for the tech debt ticket to find a better solution.

**Comment by johno1985** [comment]
This PR can now be abandoned in place of the latest changes at https://github.com/hmrc/cip-phone-number-acceptance-tests/pull/2

**Pull request closed without merging**